ProjFS.Mac/PrjFSKext/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -131,27 +193,58 @@ int16_t VirtualizationRoots_LookupVnode(vnode_t vnode, vfs_context_t context)
     return rootIndex;
 }
 
-static int16_t FindUnusedIndex_Locked()
+static VirtualizationRootHandle FindUnusedIndex_Locked()
 {
-    for (int16_t i = 0; i < MaxVirtualizationRoots; ++i)
+    for (VirtualizationRootHandle i = 0; i < s_maxVirtualizationRoots; ++i)
     {
         if (!s_virtualizationRoots[i].inUse)
         {
             return i;
         }
     }
     
-    return -1;
+    return RootHandle_None;
+}
+
+static VirtualizationRootHandle FindUnusedIndexOrGrow_Locked()
+{
+    VirtualizationRootHandle rootIndex = FindUnusedIndex_Locked();
+    
+    if (RootHandle_None == rootIndex)
+    {
+        // No space, resize array
+        uint16_t newLength = MIN(s_maxVirtualizationRoots * 2u, INT16_MAX + 1u);
+        if (newLength <= s_maxVirtualizationRoots)
+        {
+            return RootHandle_None;
+        }
+        
+        VirtualizationRoot* grownArray = Memory_AllocArray<VirtualizationRoot>(newLength);
```

Comment:
Check if `Memory_AllocArray` returned `nullptr`?

Reply:
Good catch! Added the check.